railties/lib/rails/generators/rails/app/templates/config/routes.rb.tt

```diff
@@ -6,8 +6,8 @@ Rails.application.routes.draw do
   get "up" => "rails/health#show", as: :rails_health_check
 
   # Render dynamic PWA files from app/views/pwa/*
-  get "service-worker" => "rails/pwa#service_worker", as: :pwa_service_worker
-  get "manifest" => "rails/pwa#manifest", as: :pwa_manifest
+  get "service-worker" => "rails/pwa#service_worker", as: :pwa_service_worker, constraints: {format: "js"}
+  get "manifest" => "rails/pwa#manifest", as: :pwa_manifest, constraints: {format: "json"}
```

Comment:
[As per the docs](https://guides.rubyonrails.org/routing.html#:~:text=You%20can%20use%20a%20lambda%20like%20in%20get%20%27foo%27%2C%20constraints%3A%20lambda%20%7B%20%7Creq%7C%20req.format%20%3D%3D%20%3Ajson%20%7D%20to%20only%20match%20the%20route%20to%20explicit%20JSON%20requests.), it seems like this. needs to be a lambda 
```suggestion
  get "service-worker" => "rails/pwa#service_worker", as: :pwa_service_worker, , constraints: -> { |req| req.format == :json }
  get "manifest" => "rails/pwa#manifest", as: :pwa_manifest, constraints: -> { |req| req.format == :json }
```